src/shims.h

```diff
@@ -50,11 +52,11 @@
 #endif
 #endif // HAVE_PTHREAD_WORKQUEUES
 
-#if DISPATCH_USE_INTERNAL_WORKQUEUE
+#ifdef DISPATCH_USE_INTERNAL_WORKQUEUE
 #include "event/workqueue_internal.h"
 #endif
 
-#if HAVE_PTHREAD_NP_H
+#ifdef HAVE_PTHREAD_NP_H
```

Comment:
revert

Reply:
```
error: expected value in expression
#if HAVE_PTHREAD_NP_H
```

This should be defined by CMake, correct?

Reply:
yes, also libdispatch doesn't build cleanly with `-Wundef` today, which we should likely fix long term, but moving random `#if` to `#ifdef` will break things.

Reply:
I looked further into this, and it seems ifdef is correct.

config.h.in uses #cmakedefine which does not evaluate to 1 or 0 but to defined or not defined, as stated in CMake documentation: https://cmake.org/cmake/help/latest/command/configure_file.html

Reply:
then the fix is to make it `cmakedefine01`

Darwin doesn't use CMake and we do define some of these HAVE_* things to 0 internally not ifdef/ifndef.